### PR TITLE
add --unscanned option to files:scan to triger a background scan

### PR DESCRIPTION
**Pull request by icewind1991**
cc @PVince81 @rullzer

### CODE REVIEW

**Comment by PVince81** [comment]
Should this rather be an option to occ files:scan ?

The term "background" doesn't really fit here as it's not running in cron. I know we've been using that term for the "scanner that only scans unscanned items that have a size -1". Maybe we should find a term for that.

**Comment by DeepDiver1975** [comment]
I wou

**Comment by DeepDiver1975** [comment]
I wou

**Comment by PVince81** [comment]
Setting back to "Developing".

@icewind1991 please make this an extra option for `occ files:scan`.
Not sure how to call it, `--gaps-only` ? `--lazy` ?

@Xenopathic @mmattel any suggestions ?

**Comment by Xenopathic** [comment]
As far as I can tell, background scans can have the same options as normal scans (i.e. there's no need to split them into separate commands) so I agree, there should be a `--background` or `--lazy` option to trigger the background scan.

**Comment by PVince81** [comment]
Don't call it "background". This is more about "filling in the gaps in not-yet-scanner folders"

**Comment by Xenopathic** [comment]
Ah, I see. Perhaps `--unscanned` then?

**Comment by Xenopathic** [comment]
Or `--rescan=false` (with `--rescan=true` being the current behaviour)

**Comment by mmattel** [comment]
> This is more about "filling in the gaps in not-yet-scanner folders"

what about ``--fullscan=false`` (with ``--fullscan=true`` being the current behaviour)

**Comment by icewind1991** [comment]
switched from a seperate command to an `--unscanned` option

File: apps/files/lib/Command/Scan.php [hunk 1]
```diff
@@ -86,6 +86,11 @@ protected function configure() {
 				null,
 				InputOption::VALUE_NONE,
 				'will rescan all files of all known users'
+			)->addOption(
+				'unscanned',
+				null,
+				InputOption::VALUE_NONE,
+				'only scan files which are marked as not fully scanned'
```
Comment (PVince81) [inline apps/files/lib/Command/Scan.php]: s/files/folders/ ? or both ?

**Comment by PVince81** [comment]
Tested, works :+1: 

To test, set the "size" of a folder to "-1" in the database and run:
`./occ files:scan --all --unscanned -vvvvv`

It will appear. The second time it won't.

@mmattel @Xenopathic second review ? :smile:

**Comment by Xenopathic** [comment]
Code looks good (untested) :+1:

**Comment by mmattel** [comment]
Reminder: ``--unscanned`` needs an update in the documentation

**Comment by PVince81** [comment]
@mmattel thanks! Raised here https://github.com/owncloud/documentation/issues/2435

**Pull request merged**